csm/core/controllers/s3/iam_users.py

```diff
@@ -118,9 +118,11 @@ async def patch(self):
         """
         Log.debug(f"Handling  IAM USER patch request."
                   f" user_id: {self.request.session.credentials.user_id}")
+        user_name = self.request.match_info["user_name"]
+        req_ob=await self.request.json()
         try:
             schema = IamUserCreateSchema()
-            patch_body = schema.load(await self.request.json(), unknown='EXCLUDE')
+            patch_body = schema.load({"user_name": user_name,"password":req_ob['password']}, unknown='EXCLUDE')
```

Comment:
@rohitkolapkar  Please try with this code issue
schema = IamUserCreateSchema()
        Log.debug(f"Handling create IAM USER post request."
                  f" user_id: {self.request.session.credentials.user_id}")
        **try:
            **body = await self.request.json()
            request_data = schema.load(body, unknown='EXCLUDE')**
        except ValidationError as val_err:
            raise InvalidRequest(message_args=f"Invalid request body: {val_err}")**

Reply:
@kupranay It is not working, IamUserCreateSchema() requires **user_name** and **password** fields. and in self.request.json(), i am getting only **password** field. That's why I am extracting user_name from URI path.